src/main/java/Duke.java

```diff
@@ -39,10 +42,17 @@ public static void main(String[] args) {
                     System.out.println("Got it. I've added this task:\n  " + tasks.get(tasks.size() - 1));
                     System.out.println("Now you have " + tasks.size() + " tasks in the list.");
                 } else if (command.startsWith("deadline")) {
-                    String description = command.substring(9, command.indexOf("/by")).trim();
-                    String by = command.substring(command.indexOf("/by") + 4).trim();
-                    tasks.add(new Deadline(description, by));
+                    // Parse the date and time in the format d/M/yyyy HHmm
+                    String[] parts = command.split(" /by ");
+                    if (parts.length < 2) {
```

Comment:
Suggestion: Provide an example of a suitable date format to enter e.g. 17/4/2023 2PM